File: sdks/java/connect-sdk/src/main/java/com/mobiscroll/connect/resources/Calendars.java
package com.mobiscroll.connect.resources;

import java.util.List;

import com.fasterxml.jackson.core.type.TypeReference;
import com.mobiscroll.connect.ApiClient;
import com.mobiscroll.connect.models.Calendar;

/** Calendar resource. */
public final class Calendars {

    private final ApiClient api;

    public Calendars(ApiClient api) {
        this.api = api;
    }

    /** List calendars across all connected providers. */
    public List<Calendar> list() {
        return api.get("/calendars", null, new TypeReference<List<Calendar>>() {});
    }
}

File: sdks/ruby/lib/mobiscroll/connect/resources/calendars.rb
# frozen_string_literal: true

module Mobiscroll
  module Connect
    module Resources
      class Calendars
        def initialize(_config, api_client)
          @api_client = api_client
        end

        def list
          parsed = @api_client.get('/calendars')
          return [] if parsed.nil?

          Array(parsed).map { |c| Calendar.from_h(c) }
        end
      end
    end
  end
end
